src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split(",");
+                String operation = data[0];
+                int amount = Integer.parseInt(data[1]);
+
+                if (operation.equals("supply")) {
+                    totalSupply += amount;
+                } else if (operation.equals("buy")) {
+                    totalBuy += amount;
+                }
+            }
+        } catch (IOException e) {
```

Comment:
Instead of printing the stack trace, it's better to re-throw a `RuntimeException` to allow the caller to handle it appropriately. This ensures that the error is not silently ignored and can be handled at a higher level if necessary.